### ISSUE TITLE
410 Gone Status not generated

### ISSUE THREAD

**Issue by mficzel** [neos/redirecthandler-neosadapter#31]
The creating of 410-Gone status does not work anymore. I am not sure when and how this happened. I can confirm this problem exists back to Neos 3.2 up to 4.2. It is possible that the new-ui is somehow involved but i have no setups any more without the new UI to verify this.

**Comment by mficzel** [comment]
@gerhard-boden do you know about this. As far as i see the 410 feature is not covered in the behat-tests yet but this might be quite easy.

**Comment by gerhard-boden** [comment]
Yes, should be very easy to write an behat operation that checks the statusCode of the generated redirect, pretty much the same as https://github.com/neos/redirecthandler-neosadapter/blob/master/Tests/Behavior/Features/Bootstrap/RedirectOperationTrait.php#L59 but just for the status code

**Comment by kitsunet** [comment]
Important, I feel it's rather dangerous to set this automatically as it is fully cacheable by clients (and proxies) and means if you ever create a new page with the same path segment it might not be accessible to some.

**Comment by mficzel** [comment]
In that case we should remove the feature or disable by default

**Comment by gerhard-boden** [comment]
I always deactivated this status code via configuration on my installations (so I know it worked at some point). Maybe it went away (pun intended) during the last rewrite. Nobody else seems to have noticed, but we should mention the change in behaviour since 3.0

**Comment by gerhard-boden** [comment]
Is removing this method and those lines enough?

https://github.com/neos/redirecthandler-neosadapter/blob/master/Classes/Service/NodeRedirectService.php#L285-L300
https://github.com/neos/redirecthandler-neosadapter/blob/master/Classes/Service/NodeRedirectService.php#L261-L263

(also the settings and the readme)

**Comment by gerhard-boden** [comment]
trick question: Is it a breaking change when you remove a broken feature 🤔 ?

**Comment by mficzel** [comment]
Not breaking, it is more a clarification to the docs.

**Comment by kdambekalns** [comment]
Good URLs don't change and all that… But I still think 410 has value, and is better than a 404 if something is gone… at least you know you are looking in the right place and can stop looking further.

🤷‍♂️
